lib/watchpool/watchpool.go

```diff
@@ -0,0 +1,169 @@
+package watchpool
```

Comment:
Yeah hence a separate package here to make that easy. It seemed best to experiment locally first though - `memdb.WatchSet` itself is infact standalone and not really couple to `memdb` at all too but this could live alongside later.